### PR TITLE
test: Add integration test runs as part of mainline merge and release  bump workflows

### PR DESCRIPTION
**Pull request by YutongLi291**
### What was the problem/requirement? (What/Why)
For better code quality gates, we should run the integration tests that test the Job Attachments functionality on release, as well as on merges to mainline.
### What was the solution? (How)
On PR merge to mainline, or release of the package, automated integration test workflows will be triggered. If the test workflow fails on release of the package, it will block the release.
### What is the impact of this change?
Bad code changes that break the entire Job Attachments functionality should be observable after mainline is merged into, 

Furthermore, failing tests will stop the release workflow when release bump is ran.
### How was this change tested?
`hatch build` was still working.

The integration tests do work properly as tested in https://github.com/aws-deadline/deadline-cloud/actions/runs/9765921768
### Was this change documented?
No
### Is this a breaking change?
No
----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

**Comment by leongdl** [comment]
Can we improve the "What was the problem/requirement? (What/Why)
"Setup integration tests on PR Merge to increase quality gates" - this repo is public and other users read this!

What was the solution? (How)
"Add these workflows."  - Can we make this more descriptive. EG: "On PR merge to mainline, automated integration test workflows will be triggered"

File: .github/workflows/release_bump.yml [hunk 1]
```diff
@@ -21,8 +21,31 @@ jobs:
     name: Unit Tests
     uses: ./.github/workflows/code_quality.yml
     with:
+      branch: mainline 
+  LinuxIntegrationTests:
+    name: Linux Integration Tests
```
Comment (moorec-aws) [inline .github/workflows/release_bump.yml:26]: We should add `needs: UnitTests` here. No point in running integration tests if Unit Tests are going to fail.

File: .github/workflows/release_bump.yml [hunk 2]
```diff
@@ -21,8 +21,31 @@ jobs:
...
+    uses: aws-deadline/.github/.github/workflows/reusable_integration_test.yml@mainline
+    secrets: inherit
+    with:
+      repository: ${{ github.event.repository.name }}
       branch: mainline
-
+      environment: mainline
+      os: linux
+  WindowsIntegrationTests:
+    name: Windows Integration Tests
```
Comment (moorec-aws) [inline .github/workflows/release_bump.yml:39]: Same thing here, `needs: UnitTests`.

File: .github/workflows/release_bump.yml (hunk 1 above)
Comment (YutongLi291) [reply .github/workflows/release_bump.yml:26]: Revised and fixed

File: .github/workflows/release_bump.yml (hunk 2 above)
Comment (YutongLi291) [reply .github/workflows/release_bump.yml:39]: Revised and fixed

**Review by moorec-aws — APPROVED** [review]

**Review by AlexTranAmz — APPROVED** [review]

**Pull request merged**